### PR TITLE
Feat/login

### PR DESCRIPTION
**Pull request by Renanziin-Nt**
- Implementation of Login Service: Added a new service for user authentication, including credential verification and JWT token generation.

- Environment Variables Configuration: The "SECRET_KEY" is now loaded from an environment variable, enhancing the security and flexibility of the application.

- Addition of " .env.example": Included a ".env.example" file in the repository to facilitate environment setup for other developers.

### CODE REVIEW

File: internal/v1/user/services/auth.go [hunk 1]
```diff
@@ -1 +1,48 @@
...
+	"os"
+	"time"
+)
+
+type ServiceAuth interface {
+	Login(request *dto.LoginUserRequest) (string, error)
+}
+
+type serviceAuth struct {
+	repo repository.UserRepository
```
Comment (ErickClinton) [inline internal/v1/user/services/auth.go]: aqui não é pra acessar repository.
aqui se acessa o service do user

**Review by ErickClinton — CHANGES_REQUESTED** [review]

File: internal/v1/user/services/auth.go (hunk 1 above)
Comment (Renanziin-Nt) [reply internal/v1/user/services/auth.go]: Done!

File: api/v1/user/router.go [hunk 2]
```diff
@@ -12,12 +12,12 @@ func RegisterRoutes(r *gin.Engine, db *gorm.DB) {
...
 	service := services.NewService(repo)
 	handler := NewHandler(service)
-
 	v1 := r.Group("/api/v1/users")
 	{
 		v1.POST("/", handler.create)
 		v1.GET("/:id", handler.GetByID)
 		v1.PUT("/:id", handler.UpdateById)
 		v1.DELETE("/:id", handler.Delete)
+
```
Comment (ErickClinton) [inline api/v1/user/router.go]: excluir linha

**Review by ErickClinton — APPROVED** [review]

File: api/v1/user/router.go (hunk 2 above)
Comment (Renanziin-Nt) [reply api/v1/user/router.go]: Done!

**Pull request merged**